logstash-core/src/main/java/org/logstash/config/ir/compiler/ComputeStepSyntaxElement.java

```diff
@@ -49,17 +52,27 @@
 
     private final Class<T> type;
 
+    /**
+     * Unique ID within a pipeline but should be identical across workers
+     * {@link Dataset CompiledExecution}
+     */
+    private String id;
+
     public static <T extends Dataset> ComputeStepSyntaxElement<T> create(
+        final String id,
         final Iterable<MethodSyntaxElement> methods, final ClassFields fields,
         final Class<T> interfce) {
-        return new ComputeStepSyntaxElement<>(methods, fields, interfce);
+        return new ComputeStepSyntaxElement<>(id, methods, fields, interfce);
     }
 
-    private ComputeStepSyntaxElement(final Iterable<MethodSyntaxElement> methods,
-        final ClassFields fields, final Class<T> interfce) {
+    private ComputeStepSyntaxElement(final String id,
+        final Iterable<MethodSyntaxElement> methods,
+        final ClassFields fields, final Class<T> interfce)
+    {
         this.methods = methods;
         this.fields = fields;
         type = interfce;
+        this.id = id;
```

Comment:
nitpick: I think that keeping assignments in the order they are given in args makes them easier to read

Reply:
agree.